apis/scheduling/v1alpha1/reservation_types.go

```diff
@@ -62,8 +62,29 @@ type ReservationSpec struct {
 	// +kubebuilder:default=true
 	// +optional
 	AllocateOnce *bool `json:"allocateOnce,omitempty"`
+	// AllocatePolicy represents the allocation policy of reserved resources that Reservation expects.
+	// +kubebuilder:validation:Enum=Aligned;Restricted
+	// +optional
+	AllocatePolicy ReservationAllocatePolicy `json:"allocatePolicy,omitempty"`
 }
 
+type ReservationAllocatePolicy string
+
+const (
+	// ReservationAllocatePolicyDefault means that there is no restriction on the policy of reserved resources,
+	// and allocated from the Reservation first, and if it is insufficient, it is allocated from the node.
+	ReservationAllocatePolicyDefault ReservationAllocatePolicy = ""
+	// ReservationAllocatePolicyAligned indicates that the Pod allocates resources from the Reservation first.
+	// If the remaining resources of the Reservation are insufficient, it can be allocated from the node,
+	// but it is required to strictly follow the resource specifications of the Pod.
+	// This can be used to avoid the problem that a Pod uses multiple Reservations at the same time.
```

Comment:
> This can be used to avoid the problem that a Pod uses multiple Reservations at the same time

This means that the default policy can use multiple Reservations？

Reply:
Indeed. At present, in the existing implementation of the main branch, there will be situations where the Pod uses multiple Reservations at the same time. For example, the user creates ReservationA and reserves 4C, and ReservatonB reserves 8Gi of memory, and Pod A matches the two Reservations, so that Pod A uses the resources of the two Reservations. Of course, the scheduler ensures that the entire node dimension will not be oversold.

Reply:
We can consider changing the default behavior here, such as changing to Aligned or Restricted. But here is a scenario that needs to be confirmed: If all Aligned or all Restricted on a node are not satisfied, should this node be filtered out? The current implementation in this PR is filtering out, but if the default behavior is Aligned/Restricted, this filtering behavior may not be correct. Or whether to decide to filter out this node depends on whether the user's Pod has set Reservation Affinity?

WDYT @hormes